### DISCUSSION TITLE
snips.sh behind an Ingress on K8s

### DISCUSSION THREAD

**Discussion by Abuelodelanada** [robherley/snips.sh#48] [General]
Hello, 

I would like to run snips.sh on K8s. For that I will use an ingress that provides an URL like this one for the snips.sh instance: `http://192.168.122.10/snips-snips-0/`

When I use that url in the env var `SNIPS_HTTP_EXTERNAL`, snips removes the `path` provided by the ingress (in this case: `snips-snips-0`)

For instance: 

```shell
$ cat src/charm.py | ssh 10.1.75.40 -p 2222
Pseudo-terminal will not be allocated because stdin is not a terminal.
Enter passphrase for key '/home/jose/.ssh/id_rsa': 
                                                  
┃ File Uploaded 📤                                
┃ id: gvi4MTObj0                                  
┃ size: 4.8 kB • type: python • visibility: public
                                    
┃ SSH 📠                            
┃ ssh f:gvi4MTObj0@localhost -p 2222
┃ URL 🔗                               
┃ http://192.168.122.10:80/f/gvi4MTObj0
```
the url generated is `http://192.168.122.10:80/f/gvi4MTObj0` and should be `http://192.168.122.10:80/snips-snips-0/f/gvi4MTObj0`

Is there a way to avoid snips.sh remove the `path`?

**Comment by robherley** [comment]
👋  Thanks for using snips!

Generally I'm not a huge fan of path-based ingress routing � , would very much prefer a host-based or subdomain as an alternative.

I threw together a quick edit of what it would be like to enable path based routing: https://github.com/robherley/snips.sh/compare/robherley/external-path-routing

using: `SNIPS_HTTP_EXTERNAL="http://localhost:8080/some/prefix"`

<img width="732" alt="image" src="https://github.com/robherley/snips.sh/assets/16991201/760aa100-6c6f-455e-afab-b62daa308909">

Feel free to pull that branch and try it out!

If I were to add this fully, I'd definitely want a more batteries-included router like [chi](https://github.com/go-chi/chi) and would need some actual tests for the http server. There's a lot of corner cases around routes when it comes to signing the URL for private snippets, so having tests in place would give a lot more confidence.

I should have time to implement this soon 👍

**Comment by Abuelodelanada** [comment]
Hi Rob!

Thank you very much!
I agree with you. I very much prefer a host-based or subdomain as an alternative as well.

I asked about this, because I'm writing (for fun) a Charmed Operator for snips, and the Ingress I'll use provides the alternative to use path-ingress routing.

I'll give it a try, and let you know.

Cheers
Jose
